lib/phobos/deep_struct.rb

```diff
@@ -0,0 +1,29 @@
+require 'ostruct'
+module Phobos
+  class DeepStruct < OpenStruct
+    def initialize(hash=nil)
+      @table = {}
+      @hash_table = {}
+
+      if hash
+        hash.each_pair do |k, v|
+          k = k.to_sym
+          @table[k] = if v.is_a?(Hash)
```

Comment:
I'd extract this if-elsif-else chunk into a method, the initialise is a bit hard to follow